turnpike/plugins/vpn.py

```diff
@@ -0,0 +1,86 @@
+import logging
+import re
+
+from http import HTTPStatus
+from flask import request
+
+from ..plugin import TurnpikePlugin
+
+
+class VPNPlugin(TurnpikePlugin):
+    vpn_pattern = r"(mtls\.|)private\.(console|cloud)\.(stage\.|dev\.|)redhat\.com"
```

Comment:
Very nitpicky, but I would suggest modifying the regex a bit:

```suggestion
    vpn_pattern = r"(?:mtls\.)?private\.(?:console|cloud)\.(?:(stage|dev)\.)?redhat\.com"
```

- It turns the `mtls` and the `console|cloud` groups in non-capturing groups, since we are not interested in them.
- It ignores the final `.` in the `stage|dev` non-captured, but still captures `stage` or `dev`.
- Instead of matching an empty character with `|`, it uses the `?` quantifier to make the groups optional.